File: src/main/java/dev/abarmin/aml/dashboard/model/PageModel.java
package dev.abarmin.aml.dashboard.model;

import dev.abarmin.aml.dashboard.domain.PageProps;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.ArrayList;
import java.util.Collection;

@Data
@AllArgsConstructor
public class PageModel {
  private long pageId;
  private String pageTitle;
  @Valid
  private PageProps pageProps;

  private Collection<BlockModel> pageBlocks = new ArrayList<>();
}

File: aml-frontend-private/src/model/PageModel.ts
import type { BlockResponse } from "./BlockModel";

export class PageModel {
  pageId: number;
  pageTitle: string;
  pageProps: PageProps;
  pageBlocks: BlockResponse[];

  constructor(
    pageId: number,
    pageTitle: string,
    pageProps: PageProps,
    pageBlocks: BlockResponse[]
  ) {
    this.pageId = pageId
    this.pageTitle = pageTitle
    this.pageProps = pageProps
    this.pageBlocks = pageBlocks
  }
}

export class PageProps {
  type: string;
  backgroundColor: string;

  constructor(
    type: string,
    backgroundColor: string
  ) {
    this.type = type;
    this.backgroundColor = backgroundColor;
  }
}